### PR TITLE
#60#229 Stabilize yarn build and add codestyle reporting to ci build

### PR DESCRIPTION
**Pull request by tortmayr**
Typescript
- Use reexported inversify version from Theia to ensure future stable builds
- Fix es-lint violation in `glsp-diagram-widget.ts`

Package.json
- Introduce dedicated build script for `build` and `lint`

Jenkinsfile
- Refactor build stage
- Introduce eslint stage + enable recording & publishing of issues
- Only trigger deployment job if public package code was changed 
 (part of eclipse-glsp/glsp/issues/229)

Part of: eclipse-glsp/glsp/issues/60

### CODE REVIEW

File: packages/theia-integration/package.json
Comment (planger) [inline packages/theia-integration/package.json:29]: I'm wondering whether we should either add all committers or no one here?

**Review by planger — COMMENTED** [review]
Thanks, looks good to me! :+1:
Still works as expected and it is great to have the linting checks as explicit checks in the CI. Thanks!
Just one small remark below.

File: packages/theia-integration/package.json
Comment (tortmayr) [reply packages/theia-integration/package.json:29]: Actually that's something I wanted to discuss in the next meeting. IMO it is quite tedious to maintain a list of all committers here. However, omitting it entirely might also be contra productive because it we probably need this list for sponsored and/or external contributions.
So I'd propose that we initially only list the package creator here. External contributors are free to extend this list.
Alternatively we could also use a generic author like:
```json
    "name": "Eclipse GLSP",  
    "email": "<someGenericMailAddress>",   
    "url": "https://www.eclipse.org/glsp/"
```
or 

```json
    "name": "EclipseSource",  
    "email": "<mailOfPackageCreator>",   
    "url": "https://eclipsesource.com/"
```

The same also has to be considered for the `<developers>` section in pom.xml files.

If we follow the Theia/Sprotty pattern of not maintaining contributor lists in the copyright header we then have the following possibilities to mention external contributors/sponsors

- The commit message
- The changelog
- The corresponding section in the package.json or pom.xml

Comment (planger) [reply packages/theia-integration/package.json:29]: Ah ok I see, thanks! Let's discuss this in a separate issue then and not block this change.
Thanks!

**Review by planger — APPROVED** [review]
Thanks, looks good to me! :+1:

File: packages/theia-integration/package.json
Comment (tortmayr) [reply packages/theia-integration/package.json:29]: I created https://github.com/eclipse-glsp/glsp/issues/246 to track this

**Pull request merged**
